libs/orchestrator-ui-components/src/lib/components/Table/Table.tsx

```diff
@@ -1,140 +1,89 @@
-        if (!dataRow) {
-            return;
-        }
-
-        const column = columns[columnId as keyof T];
-        const cellValue = dataRow[columnId as keyof T];
-
-        handleRowClick &&
-            setCellProps({
-                css: { cursor: 'pointer' },
-                onClick: () => handleRowClick(dataRow),
-            });
-
-        return column.renderCell
-            ? column.renderCell(cellValue, dataRow)
-            : `${cellValue}`;
-    };
-
-    const controlColumnToEuiDataGridControlColumnMapper: (
-        controlColumn: ControlColumn<T>,
-    ) => EuiDataGridControlColumn = ({ id, width, rowCellRender }) => ({
-        id,
-        width,
-        headerCellRender: (props) => null,
-        rowCellRender: ({ rowIndex }) => {
-            const { pageSize, pageIndex } = pagination;
-            const rowIndexOnPage = rowIndex - pageIndex * pageSize;
-
-            const dataRow = data[rowIndexOnPage];
-
-            return rowCellRender(dataRow);
-        },
-    });
-
-    const euiDataGridLeadingControlColumns = leadingControlColumns?.map(
-        controlColumnToEuiDataGridControlColumnMapper,
-    );
-
-    const euiDataGridTrailingControlColumns = trailingControlColumns?.map(
-        controlColumnToEuiDataGridControlColumnMapper,
-    );
-
-    const gridHeightValue =
-        pagination.pageSize * 40 + EUI_DATA_GRID_HEIGHT_OFFSET;
-
-    return (
-        <EuiDataGrid
-            aria-label="Data Grid"
-            columns={initialColumnOrderRef.current}
-            leadingControlColumns={euiDataGridLeadingControlColumns}
-            trailingControlColumns={euiDataGridTrailingControlColumns}
-            height={`${gridHeightValue}px`}
-            gridStyle={GRID_STYLE}
-            columnVisibility={{ visibleColumns, setVisibleColumns }}
-            pagination={pagination}
-            sorting={columnSortToEuiDataGridSorting(
-                dataSorting,
-                updateDataSorting,
-            )}
-            rowCount={pagination.totalRecords}
-            renderCellValue={renderCellValue}
-        />
-    );
-};
+    pagination,
+    isLoading,
+    onCriteriaChange,
+    onDataSort,
+}: TableProps<T>) => (
+    <EuiBasicTable
+        tableCaption="Demo of EuiBasicTable"
+        items={data}
+        rowHeader="firstName"
+        columns={mapTableColumnsToEuiColumns(
+            columns,
+            hiddenColumns,
+            dataSorting,
+            onDataSort,
+        )}
+        pagination={pagination}
+        onChange={onCriteriaChange}
+        loading={isLoading}
+    />
+);
+
+function mapTableColumnsToEuiColumns<T>(
+    columns: TableColumns<T>,
+    hiddenColumns?: Array<keyof T>,
+    dataSorting?: DataSorting<T>,
+    onDataSort?: (columnId: keyof T) => void,
+): EuiBasicTableColumn<T>[] {
+    return Object.keys(columns)
+        .filter((colKey) => !hiddenColumns?.includes(colKey as keyof T))
+        .map((colKey): EuiBasicTableColumn<T> => {
```

Comment:
Just wondering if it would make sense to extract these functions (in filter and map) instead of inlining everything as a lambda since they are pretty big.

Reply:
Yes, that might be better. Updated, I kept the extracted functions within the mapper